assignment3/assignment3.cpp

```diff
@@ -6,191 +7,158 @@
 #include <set>
 
 class Dictionary{
-	std::vector<std::string> dictionary;
+    std::vector<std::string> dictionary;
 public:
-	Dictionary(const std::string &filemame){
-		std::ifstream file (filemame);
-  		if (file.is_open()) {
- 			std::string line;
-    		while (std::getline (file,line)) {
-      			dictionary.push_back(line);
-    		}
-    		file.close();
-  		}
-	}
-	bool isPrefix(const std::string &word) const {
-		bool is_prefix = false;
-		for(int i = 0; i<dictionary.size() && !is_prefix; i++){
-			is_prefix = true;
-			for(int j = 0; j<word.size() && is_prefix; j++){
-				if(j<dictionary[i].size()){
-					is_prefix = dictionary[i][j] == word[j];
-				} else {
-					is_prefix=false;
-				} 
-			}
-		}
-		return is_prefix;
-
-	}
-	bool isWord(const std::string &word) const{
-		bool is_word = false;
-		for(int i = 0; i<dictionary.size() && !is_word; i++){
-			is_word = dictionary[i].compare(word) == 0;
-		}
-		return is_word;
-	}
+    Dictionary(const std::string &filemame){
+        std::ifstream file (filemame);
+        if (file.is_open()) {
+            std::string line;
+            while (std::getline (file,line)) {
+                dictionary.push_back(line);
+            }
+            file.close();
+        } else {
+            dictionary.clear();
+        }
+    }
+    bool isPrefix(const std::string &word) const {
+        bool is_prefix = false;
+
+        for(int i = 0; i<dictionary.size() && !is_prefix; i++){
+            is_prefix = word.compare(dictionary[i].substr(0,word.size())) == 0;
```

Comment:
The idea was to use one not necessarily both :)
I think this version of compare is equivalent to using `==`, i.e: 
`word == dictionary[i].substr(0,word.size())`
Or using another form of compare:
`word.compare(0, word.size(), dictionary[i]) == 0`